src/loader/networks/sdex-chain.ts

```diff
@@ -28,7 +45,182 @@ export class SdexChain {
+      return balance.toString();
+    }
+  }
+
+  private async updateLPTokens(baseToken: string, quoteToken: string) {
+    if (!this.lpTokens.includes(baseToken)) {
+      this.lpTokens.push(baseToken);
+    }
+    if (!this.lpTokens.includes(quoteToken)) {
+      this.lpTokens.push(quoteToken);
+    }
+  }
+
+  public async getUpdatedLiquidity(user: string, base: string, quote: string, poolIdx: number) {
+    const { liquidityChanges } = await this.queryUserPositions(user);
+
+    const concentratedPositions = filterPositions(liquidityChanges, poolIdx, base, quote, PositionType.concentrated);
+    const ambientPositions = filterPositions(liquidityChanges, poolIdx, base, quote, PositionType.ambient);
+
+    if (concentratedPositions.length === 0 && ambientPositions.length === 0) {
+      return null;
+    }
+
+    await this.updateLPTokens(base, quote);
+
+    const ambientPositionResults = await Promise.all(
+      ambientPositions.map(async (userLiquidity) => {
+        const ambientTokens = await this.queryAmbientTokens(user, base, quote, poolIdx);
+        const [lpTokenBalanceBase, lpTokenBalanceQuote] = await Promise.all([
+          this.getLPTokenBalance(user, base),
+          this.getLPTokenBalance(user, quote),
+        ]);
+        const ambientLiq = bignumber(ambientTokens.liq)
+          .plus(bignumber(lpTokenBalanceBase))
+          .plus(bignumber(lpTokenBalanceQuote))
+          .toString();
+
+        return {
+          ambientLiq,
+          concLiq: '0',
+          rewardLiq: '0',
+          baseQty: ambientTokens.baseQty.toString(),
+          quoteQty: ambientTokens.quoteQty.toString(),
+          aggregatedLiquidity: userLiquidity.liq,
+          aggregatedBaseFlow: userLiquidity.baseFlow,
+          aggregatedQuoteFlow: userLiquidity.quoteFlow,
+          positionType: userLiquidity.positionType,
+        };
+      }),
+    );
+
+    const aggregatedAmbientPosition = aggregatePositions(ambientPositionResults);
+
+    const concentratedPositionsResults = await Promise.all(
+      concentratedPositions.map(async (userLiquidity) => {
+        const { bidTick, askTick } = userLiquidity;
+        const [rangeTokens, rewardLiq] = await Promise.all([
+          this.queryRangeTokens(user, base, quote, poolIdx, bidTick, askTick),
+          this.queryConcRewards(user, base, quote, poolIdx, bidTick, askTick),
+        ]);
+
+        if (!bignumber(rangeTokens.liq).isZero()) {
+          return {
+            ambientLiq: '0',
+            concLiq: rangeTokens.liq.toString(),
+            rewardLiq: rewardLiq.liqRewards.toString(),
+            baseQty: rangeTokens.baseQty.toString(),
+            quoteQty: rangeTokens.quoteQty.toString(),
+            aggregatedLiquidity: userLiquidity.liq.toString(),
+            aggregatedBaseFlow: userLiquidity.baseFlow.toString(),
+            aggregatedQuoteFlow: userLiquidity.quoteFlow.toString(),
+            positionType: userLiquidity.positionType,
```

Comment:
`bidTick` and `askTick` must be exposed in the list. It's being used to calculate current user liquidity in the FE.

Reply:
We also need these exposed too:

"aprDuration": 4632316,
"aprPostLiq": 32770858.742432915,
"aprContributedLiq": 32761856,
"aprEst": 0.001872239175168433,

current indexer example: https://bob-ambient-graphcache.sovryn.app/gcgo/user_pool_positions?user=0x2bd2201bfe156a71eb0d02837172ffc237218505&base=0x05d032ac25d322df992303dca074ee7392c117b9&quote=0xe75d0fb2c24a55ca1e3f96781a2bcc7bdba058f0&poolIdx=400&chainId=0xed88